### ISSUE TITLE
Over 90% of `normalize` disappears when building an app in production mode

### ISSUE THREAD

**Issue by artisologic** [styled-components/polished#262]
# TLDR
I believe #253 has introduced unwanted `/*#__PURE__*/` annotations on certain lines of code, resulting in `UglifyJS` deleting said lines when building in production mode.

If you want to follow the process I went through to get to this, continue reading 👇

---

I seem to have hit a very weird issue that only occurs when I build my app in production mode.
I am using a `create-react-app` generated app, so I've included `react-scripts`'s version in the relevant section below:

- `polished` version: **1.8.1**
- `JSS-in_CSS` library and version: **irrelevant**
- Any relevant JS library and version: **react-scripts 1.0.14**

### Mixin/Helper/Shortand you were using and how you were using it:

```javascript
export default function() {
	return injectGlobal`
		${normalize()};
	`;
}
```

### What you are seeing:

Only a very small subset of the `normalize` rules is injected when the app is built in production mode:

![image](https://user-images.githubusercontent.com/1539897/32024243-b781e11e-b9d3-11e7-9c18-3fb3ac5ba04a.png)

### What you expected to see:

The full version of `normalize` is injected correctly when running the app in development mode:

![image](https://user-images.githubusercontent.com/1539897/32024133-621942b2-b9d3-11e7-8299-b6d62efcf243.png)

### Reproduction:

- Generate an app with `create-react-app`.
- Use the `normalize` mixin (logging the results of the function is enough to demonstrate the issue)
- Run in dev mode using `npm start` = 👍 (logs all normalized rules)
- Build in production mode using `npm run build` = 👎 (logs very few normalized rules)

### What I think the issue is:

- The issue started appearing once we upgraded `polished` from `1.7.0` to `1.8.1`.
- After doing a diff between these 2 tags, I couldn't see anything remotely touching the `normalize` mixin.
- I then attempted to look inside the file our module loader was loading, which for us seems to be: `dist/polished.es.js`.
- I noticed in that file that all the rules I see disappear in production all have a `/*#__PURE__*/` comment before the call to `defineProperty(...)` as you can see here:

![image](https://user-images.githubusercontent.com/1539897/32024738-81a0bf64-b9d5-11e7-866e-8e7ba59b7b56.png)
- I then tested to manually remove some of these, and indeed the rules re-appeared in production.
- Once I knew this seemed to be the issue I went back to the diff between `1.7.0` to `1.8.1` and it seems this PR is to blame: #253.
- It looks like it introduced an automated way to add these annotations via a babel plugin and remove the ones that existed manually in the code.
- Looking at that [babel plugin](https://www.npmjs.com/package/babel-plugin-annotate-pure-calls) I understood it had something to do with helping UglifyJS to perform dead-code elimination and guessed that in our case it was somewhat being too greedy and deleting these lines of code.
- I tested that by manually updating the production webpack config in `react-scripts` and disabling `UglifyJS`. Indeed all rules came back.
- **So my guess is that #253 has introduced unwanted `/*#__PURE__*/` annotations on certain lines of code, resulting in `UglifyJS` deleting said lines when building in production mode.**

**Comment by bhough** [comment]
@artisologic Thanks for taking the time to provide all these details. I will take a look at this tonight and see if I can work with @Andarist to come up with a quick solution.

**Comment by Andarist** [comment]
One quick fix for this in polished itself would be to use `loose` mode for the babel transpilation (which I think should be used like most of the time). It won't use calls to `defineProperty` when transpiling computed keys but rather simple assignments - it will also be slighlty more performant.

Of course Im happy to provide a fix for this in my plugin too, just need to figure out what's the best option. Any ideas/preferred solutions? Im thinking about:
- blacklisting `defineProperty` (easiest)
- recognizing default computed properties pattern and transforming it into something which won't introduce such issue - i.e. nested `defineProperty` calls as it return the defined object anyway, that would actually minimize the output by a little bit too
- implementing this ^ in babel's computed properties plugin - although would need to consider speced order order of calls, but i think it's possible to do that within spec constraints

**Comment by bhough** [comment]
@artisologic `1.8.2` is now out with a move to loose mode that should fix this. Please let me know if you run into any further issues.

**Comment by artisologic** [comment]
@bhough Great! Thanks for the fix, I will test that out tomorrow and report.

**Comment by artisologic** [comment]
@bhough I can confirm it's fixed the issue for us. Thanks!
